File: solutions/java/grains/1/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger grainsOnSquare(final int square) {
        if (square < 1 || square > 64) {
        throw new IllegalArgumentException("square must be between 1 and 64");
    }
        return BigInteger.TWO.pow(square-1);
    }
    BigInteger grainsOnBoard() {
        BigInteger totalGrain = new BigInteger("0");
        for(int square = 1; square<65; square++){
            totalGrain = totalGrain.add(grainsOnSquare(square));
        };
        return totalGrain;
    }

}

File: solutions/python/grains/1/grains.py
def square(n):
    if n<=0 or n>64:
        raise ValueError("square must be between 1 and 64")
    return 2**(n-1)


def total():
    result = 0;
    for x in range(1,65):
        result += square(x)
    return result
